moped-editor/src/views/projects/projectView/ProjectComponents/ComponentListItem.js

```diff
@@ -8,73 +7,34 @@ import ListItem from "@material-ui/core/ListItem";
 import ListItemSecondaryAction from "@material-ui/core/ListItemSecondaryAction";
 import ZoomInIcon from "@material-ui/icons/ZoomIn";
 import IconButton from "@material-ui/core/IconButton";
-import DeleteIcon from "@material-ui/icons/Delete";
-import Button from "@material-ui/core/Button";
-import { EditOutlined } from "@material-ui/icons";
+import PlaceOutlinedIcon from "@material-ui/icons/PlaceOutlined";
 import ListItemText from "@material-ui/core/ListItemText";
 import { COLORS } from "./mapStyleSettings";
+import { useComponentListItemText } from "./utils/componentList";
 
 const useStyles = makeStyles((theme) => ({
   nested: {
     paddingLeft: theme.spacing(4),
   },
+  listItem: {
+    paddingLeft: theme.spacing(1),
+  },
+  listItemText: {
+    marginLeft: theme.spacing(1),
+  },
 }));
 
-const isSignalComponent = (component) =>
-  component?.moped_components?.feature_layer?.internal_table ===
-  "feature_signals";
-
-/**
- * Format the list item's primary and secondary text
- * @param {object} component - the moped_component object
- * @returns {object} the component props for <ListItemText> as { primary, secondary }
- */
-const useComponentListItemText = (component) =>
-  useMemo(() => {
-    const listItemText = { primary: "", secondary: "" };
-    const componentName = component?.moped_components?.component_name;
-    const componentSubtype = component?.moped_components?.component_subtype;
-    listItemText.primary = componentSubtype
-      ? `${componentName} - ${componentSubtype}`
-      : componentName;
-    if (isSignalComponent(component)) {
-      const signalLocationName = component?.feature_signals?.[0]?.location_name;
-      const signalId = component?.feature_signals?.[0]?.signal_id;
-      listItemText.secondary = `${signalId}: ${signalLocationName}`;
-    }
-    return listItemText;
-  }, [component]);
-
 export default function ComponentListItem({
   component,
   isExpanded,
-  setClickedComponent,
-  setIsDeletingComponent,
-  editDispatch,
-  onClickZoomToComponent,
-  isEditingComponent,
-  isCreatingComponent,
```

Comment:
I refactored `ComponentListItem` to be general enough to use for the current project's components and its related projects too.